src/main/java/com/aws/greengrass/deployment/ShadowDeploymentListener.java

```diff
@@ -308,6 +314,24 @@ protected void shadowUpdated(Map<String, Object> desired, Map<String, Object> re
                             .log("Deployment result already reported. Ignoring shadow update at startup");
                     return;
                 }
+                // Ignore if it's the ongoing deployment. This can happen if the last shadow deployment caused restart
+                try {
+                    // Using locate instead of injection here because DeploymentService lacks usable injection
+                    // constructor. Same as in IotJobsHelper.evaluateCancellationAndCancelDeploymentIfNeeded
+                    GreengrassService deploymentServiceLocateResult =
```

Comment:
you don't need to do this, you should be able to just `@inject` the deployment service itself.

Reply:
Sadly, directly `@inject DeploymentService` gives no usable injection constructor error